pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -297,27 +297,69 @@ export class ExtensionImpl implements Extension {
     }));
   }
 
-  get containerName() {
+  protected async getComposeName() {
+    if (this._composeName) {
+      return this._composeName;
+    }
+
     const normalizedId = this.id.toLowerCase().replaceAll(/[^a-z0-9_-]/g, '_');
+    const contents = await this.getComposeFileContents();
+    let composeName = `rd-extension-${ normalizedId }`;
+
+    const maxServiceLength = Math.max(...Object.keys(contents.services).map(n => n.length));
+
+    // On nerdctl, container names are something like:
+    // <this.composeName>_<compose .services.*>_<counter>
+    // If this string is longer than 76 characters, installation will fail.
+    if (composeName.length + maxServiceLength + 4 > 76) {
+      composeName = normalizedId.slice(0, 76 - maxServiceLength - 4);
+    }
+
+    this._composeName = composeName;
 
-    return `rd-extension-${ normalizedId }`;
+    return composeName;
   }
 
-  protected async installContainers(workDir: string, metadata: ExtensionMetadata): Promise<void> {
-    const composeDir = path.join(workDir, 'compose');
-    let contents: any;
+  /** memoized result of getComposeName() */
+  protected _composeName = '';
+
+  /**
+   * Return the contents of the compose file.
+   */
+  protected async getComposeFileContents(): Promise<any> {
```

Comment:
Is there a reason why this returns `Promise<any>`? Why couldn't we return a type?

Reply:
We don't have a type defined here, that's all. I can go define one (file-local), sure.